IBPSA/Electrical/BaseClasses/PV/BaseClasses/PVOptical/AirMass.mo

```diff
@@ -0,0 +1,43 @@
+within IBPSA.Electrical.BaseClasses.PV.BaseClasses.PVOptical;
+block AirMass
+  "Air mass calculation depening on zenith angle and height of object"
+  extends Modelica.Blocks.Icons.Block;
+  parameter Modelica.Units.SI.Height alt "Height of object";
+
+  Modelica.Blocks.Interfaces.RealInput zenAng(final unit="rad", final displayUnit="deg") "Zenith angle for object"
+    annotation (Placement(transformation(extent={{-140,-20},{-100,20}})));
+  Modelica.Blocks.Interfaces.RealOutput airMas(final unit="1")
+    annotation (Placement(transformation(extent={{100,-10},{120,10}})));
+protected
+  Modelica.Units.SI.Angle zen "Zenith angle internal use";
+equation
+// Restriction for zenith angle
+ zen = if zenAng <= Modelica.Constants.pi/2 then
+ zenAng
+ else
+ Modelica.Constants.pi/2 "Zenith angle";
+
+  airMas =
+  exp(-0.0001184*alt)/(cos(zen) + 0.5057*(96.080 -
+                  zen*180/Modelica.Constants.pi)^(-1.634)) "Air mass";
```

Comment:
Please format text for easier readability.

Reply:
done